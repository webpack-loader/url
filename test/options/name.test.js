const { webpack } = require('@webpack-utilities/test')

describe('Options', () => {
  describe('name', () => {
    test('{String}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            name: '[hash].[ext]'
          }
        }
      }

      const stats = await webpack('fixture.js', config)
      const { source } = stats.toJson().modules[0]

      expect(source).toMatchSnapshot()
    })

    test('{Function}', async () => {
      const config = {
        loader: {
          test: /(png|jpg|svg)/,
          options: {
            name() {
              return '[hash].[ext]'
            }
          }
        }
      }

      const stats = await webpack('fixture.js', config)
      const { source } = stats.toJson().modules[0]

      expect(source).toMatchSnapshot()
    })
  })
})
