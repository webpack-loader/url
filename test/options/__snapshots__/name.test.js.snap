// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options name {Function} 1`] = `"export default __webpack_public_path__ + \\"39f5c21c.png\\";"`;

exports[`Options name {String} 1`] = `"export default __webpack_public_path__ + \\"39f5c21c.png\\";"`;
